uint16 hero
uint16 engineer
uint16 infantry1
uint16 infantry2
uint16 reserve
uint16 sentry